File: pset1/cash/cash.c
// First problem of CS50 edX course
// Also: my first-ever program in C
// The porpouse of this is to count how many coins will be given to the client as change

#include <stdio.h>
#include <cs50.h> // Special Library from the course cs50, responsible for the get_float() funciton

int main(void)
{
    while (true)
    {
        int cents = 1000 * get_float("Change owed:"); // takes the value in float that has to be 
        if (cents > 0) // verifies if the inputed value is valid
        {
            // the store only has coins of 25, 10, 5 and 1 cents
            // creates a var that holds the total of coins
            int coins = 0;

            coins += cents / 250; // adds to 'coins' a int amount of "25 cents" coins
            cents = cents % 250; // keeps whats left

            // do the same for the "10 cents" coins
            coins += cents / 100;
            cents = cents % 100;

            // repeat for the "5"
            coins += cents / 50;
            cents = cents % 50;

            // whats is the change in 1's 
            coins += cents / 10;

            // prints the result: total count of coins
            printf("%i\n", coins);
            break;
        }
    }
}

File: pset6/cash/cash.py
from cs50 import get_float

coins = 0 # .50 .25 .10 .05 .01
change = get_float("Change owed: ")

if change > .50:
    coins += int(change / .50)
    change = change % .50
if change > .25:
    coins += int(change / .25)
    change = change % .25
if change > .10:
    coins += int(change / .10)
    change = change % .10
if change > .05:
    coins += int(change / .05)
    change = change % .05
if change > .01:
    coins += int(change / .01)

print(str(coins))
